### PR TITLE
STAC admin plugin system

### PR DESCRIPTION
**Pull request by danielfdsilva**

### CODE REVIEW

File: adr/0001-stac-admin-plugin-system.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+## Decision Drivers
+
+The chosen option should allow:
+- Each instance to define what fields the form editor should have;
+- Within an instance, each collection/item to define what fields the form editor should have based on some sort of, possibly dynamic, criteria;
+
+## Decision
+
+The structure of the form is defined by a plugin system. Each plugin is responsible for a section of the editor and defines the fields that should be shown.
```
Comment (emmanuelmathot) [inline adr/0001-stac-admin-plugin-system.md:20]: Will there be a plugin for each STAC extension to support (easier to identify and maintain) or could a plugin be in charge of a set of extensions ?

Comment (danielfdsilva) [reply adr/0001-stac-admin-plugin-system.md:20]: I thinks that ideally is one plugin per extension, but this is not a rigid requirement.
And even if we do a plugin per extensions, we could then bundle plugins together in a way they make sense, like "common extensions".

File: adr/0001-stac-admin-plugin-system.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+## Context and Problem Statement
+
+The form to edit the STAC metadata may differ from instance to instance and from collection to collection. Ideally the form should be easily customizable and extensible to suit the different needs.
+
+## Decision Drivers
+
+The chosen option should allow:
+- Each instance to define what fields the form editor should have;
+- Within an instance, each collection/item to define what fields the form editor should have based on some sort of, possibly dynamic, criteria;
```
Comment (oliverroick) [inline adr/0001-stac-admin-plugin-system.md:16]: I think, there should be some dependency on the extensions specified for the catalog/collection/item. You would want to support data editing for the fields specified in the extensions, within reason, as discussed, we want to restrict some fields. But as a general rule, I would aim to support as many fields as we can and the aim is to support the extensions of the catalog/collection/item.

Comment (danielfdsilva) [reply adr/0001-stac-admin-plugin-system.md:16]: I agree. I only meant that, ultimately, it is the instance to define which fields are editable without being forced to anything. The plugins we'd provide support the majority of fields, but if the instances choose to use custom ones, they're free to do so. Maybe this should be written differently to make it clearer? What do you think?

Comment (oliverroick) [reply adr/0001-stac-admin-plugin-system.md:16]: Maybe something like this:

> - Each instance controls what fields are available in the editor.
> - We will provide and maintain plugins to that support editing meta data specified by a variety of commonly used STAC extensions. 
> - Through instance-specific plugins, instances can define additional custom fields if required.

**Review by j08lue — APPROVED** [review]
I added a suggestion for an extra context/intro paragraph. Please review and consider adapting it, @danielfdsilva.

Are there any alternatives to a plugin system that would be worth mentioning, if only for completeness?

Otherwise, I think this small ADR should be considered accepted.

File: adr/0001-stac-admin-plugin-system.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+With a plugin system the editor can be easily customized to suit the needs of each instance. Implementors have the option to use a pre-made solution and have something up and running quickly or to develop their own plugins with editing widgets that suit their needs.
+
+#### Cons
+Having plugins define a schema, inevitably results in some duplication, given that the schema for the different extensions already exists defined in the STAC spec. However this schema may not be a one-to-one match to the desired form structure. Additionally, implementing a full JSON schema renderer would be a significant amount of work and would likely be overkill for the needs of the editor.
+
+Nevertheless, this can be mitigated, by having the plugin query the STAC spec on initialization to get a base schema from which to derive the editor schema.
+
+## More Information
+
+The initial proposal pro the plugin system was outlined in the [Flexible plugin system for the STAC metadata editor](https://github.com/EOEPCA/data-access/issues/73) issue.
```
Comment (danielfdsilva) [inline adr/0001-stac-admin-plugin-system.md:39]:
```suggestion
The initial proposal for the plugin system was outlined in the [Flexible plugin system for the STAC metadata editor](https://github.com/EOEPCA/data-access/issues/73) issue.
```

**Pull request merged**